content/blog/metrics-at-imperial.md

```diff
@@ -0,0 +1,121 @@
+
+### User Stories, user focus *(or Who are we doing it for?)*
+Following on from the issue of recording what we do is the question of how we record tasks. As developers it seems 
+natural to express a task in terms of the actual code changes that are made (writing a class, adding an endpoint, 
+changing configuration etc). These details are often vital to record when features are being planned, to record 
+technical solutions as they are thought of - but they are lousy as a record of what benefit we are adding for users of 
+our systems. 
+
+There's nothing very radical about this thought - it's the reason why the User Story was a key part of early 
+incarnations of [Agile development](https://www.agilealliance.org/glossary/user-stories) in the 1990s. However we do 
+recognise that as a technically-driven team, it can be easy for us to leave the user stories aside as a given when we 
+first dive into thinking about technical solutions  - which makes it harder to look back at them afterwards, and 
+remember what they were solutions for! So we intend to record all features as user stories as far as possible.
+
+### Cost and Benefit *(or Why are we doing it?)*
+Having zeroed in on the user benefit of a task, another question arises: is it really worth it? Sometimes the priority 
+we should assign to a task is hard to determine, particularly when we have many 'clients' of different kinds - 
+individual researchers of varying seniority and requirements, users of our 'products', large projects spanning multiple 
+organisations - and requests ranging from tiny cosmetic changes to entire applications. 
+
+A 'cost/benefit' metric at the task level might help us to decide which should take priority, and give us visibility of 
+why we have chosen to prioritise the tasks we have. We decided that for our needs, it would be overkill to set a strict 
+formula to calculate a numerical priority value, but that it would be useful to record a checklist for incoming tasks, 
+recording things like:
+
+Costs: 
+
+* Estimated story points (or days effort)
+* Additional effort to fully specify requirements
+* Any additional software or hardware costs
+* Any risks 
+
+Benefits:
+
+* Benefit to user (e.g. major new feature users have been asking for, or just a cosmetic improvement)
+* Urgency (Is this a 'showstopper'?)
+* How many users this will benefit, and for how long
+* Is this something which could be re-used / give future benefit to the team, or other areas of our work?
+
+All these values will be subjective and uncertain, and it may take some time for us to settle on the right checklist to 
+capture. However, having a framework such as this will enable us to start to quantify the priority of our incoming tasks 
+and give visibility to why prioritisation decisions have been made. 
+
+### Communicate Success *(or Wow!)*
+If the ultimate purpose of metrics is to prove our worth, and to enable us to improve, then it's not enough to just 
```

Comment:
Maybe this should be something like communicate successes and failures, if the goal is honest metrics rather than self-justification. Just a bit wary that "prove our worth" sounds a bit like we've already decided what the metrics should show before we've recorded them ;)

Reply:
Yeah, I tried to bring in that point when talking about communicating the stuff that didn't work too, but could maybe make it sound a bit less like it's assuming we can pat ourselves on the back! "Communicate success" was one of the four action points we came away with. "Communicate experience and achievements" might be less self-congratulatory...